packages/core/src/shareback/ensure-ccd-exists.ts

```diff
@@ -39,6 +47,10 @@ export async function ensureCcdExists({
     log,
   });
 
-  log("CCD generated. Fetching the document contents");
+  // We do this because we had situations where even though we await on the creation of the empty CCD,
+  // after this function we try to read the metadata file and S3 returned it didn't exist yet.
```

Comment:
Just perusing for educational purposes, and saw this comment - this was actually a known issue with S3 for many years, where performing a HEAD followed by an immediate object write would indicate that the object still (incorrectly) does not exist. Ostensibly this was some kind of caching issue, but in 2020, S3 finally announced [strong consistency for HEAD requests](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Welcome.html#ConsistencyModel). Perhaps the outdated SDK version of S3Utils is causing this to remain an issue? In any case, this is an instructive solution.

Reply:
done

Reply:
Actually, we do have an issue where the request for the CCD comes before the CCD is generated, so I'll have to add a retry w/ backoff there.